library/database/class.mysqlstructure.php

```diff
@@ -572,6 +572,32 @@ protected function _modify($Explicit = false) {
                 $Comment = "-- Existing: $ExistingColumnDef, New: $ColumnDef";
 
                 if ($ExistingColumnDef !== $ColumnDef) {
+                    if (strcasecmp($ExistingColumn->Type, 'varchar') === 0&& strcasecmp($Column->Type, 'varchar') === 0
+                            && $ExistingColumn->Length > $Column->Length) {
+
+                        $charLength = $this->executeQuery("select max(char_length(`$ColumnName`)) as MaxLength from `$Px{$this->_TableName}`;")
+                            ->firstRow(DATASET_TYPE_ARRAY);
+
+                        if ($charLength['MaxLength'] > $Column->Length) {
+                            $this->addIssue("The table's column was not altered because it contains a varchar of length {$charLength['MaxLength']}.", $Comment);
+
+                            // Log an event to be captured and analysed later.
+                            Logger::event(
+                                'structure_integrity',
+                                Logger::ALERT,
+                                "Cannot modify {tableName}'s column {column} because it has a value of {maxVarcharLength,number} and the new length it {rowThreshold,number} threshold.",
```

Comment:
Might want to think about rephrasing this error message. Example:

> Cannot modify GDN_Discussion's column ForeignID because it has a value of 32 and the new length it 24 threshold.